src/logmessage.cpp

```diff
@@ -122,7 +122,11 @@ namespace g3 {
 
 
    std::string LogMessage::timestamp(const std::string& time_look) const {
-      return  localtime_formatted(_timestamp, time_look);
+      std::ostringstream stringStream;
+
+      stringStream << _timestamp.tv_nsec;
+
+      return g3::localtime_formatted(_timestamp.tv_sec, (time_look.empty() ? g3::internal::date_formatted + " " + g3::internal::time_formatted : time_look)) + "|" + stringStream.str() + "|";
```

Comment:
I think it might be tricky to provide the formatting together with the nanoseconds. Let's keep the first go at this as simple as possible. 

For now,   I would keep it as it was (time_t argument) and provide a few simple functions instead that can be used by the sinks and of course by the default sink.

1. In logmessage .h/cpp  just a wrapper to return std::time_t from the internal time representation (timespec). I.e it can be converted easily on the Linux side and I can later provide the code for converting it on Windows side.

2. In logmessage .h/cpp a simple, convenience, function to return the nanosecond part as a) value 
size_t nanosecond() {...}


Then it is trivial for the default sink as well as other sinks to continue using exactly the same API as before but with the option of adding the nanoseconds information. 

If a sink wants to format his in a strange funky way (hour / nanosecond / second / year) or whatnot then the writer of that creative sink can deal with that mess.